src/lib/logstore/log_dev.cpp

```diff
@@ -259,6 +259,18 @@ int64_t LogDev::append_async(logstore_id_t store_id, logstore_seq_num_t seq_num,
     return idx;
 }
 
+log_buffer LogDev::read_from_cache(const logid_t& idx) {
```

Comment:
io_blob is NOT a safe buffer.

This implementation doesnt seems right,  the client of LogStore can probably release the byte buffer provided, before truncation.  unless we change from top-down to make it `io_blob_safe`.  

But in general I dont think this is right approach as it will consume a huge amount of memory, if client is lazily truncate.

Reply:
ignore this, we will only buffer log entry for home raft log store. so no changes will not be made.